webrtc.html

```diff
@@ -509,33 +509,47 @@
 
         <ol>
           <li>
-            <p>Append an object representing the current call being handled
-            (i.e. function name and corresponding arguments) to the
-            <code>operations</code> array.</p>
+            <p>If <var>connection</var>'s <a href=
+              "#dom-peerconnection-signaling-state"><code>RTCPeerConnection</code>
+              signalingState</a> is <code>closed</code>, throw an
+              <code>InvalidStateError</code> exception and abort these steps.</p>
+          </li>
+
+          <li>
+            <p>If <code>operations</code> is empty, execute the
+              function and let <var>p</var> be the promise returned by
+              the function.  If <code>operations</code> is non-empty,
+              let <var>p</var> be a promise equivalent to the promise
+              that would be returned by calling the function (it will
+              be settled at the same time and with the same result).
```

Comment:
Only one of the promises is seen by JS, so they don't have to be at the same time "in Javascript".  The one returned is simply a proxy for the one that doesn't exist yet.

It doesn't matter how you implement it as long as the order of the events is the same.  You can use promise chains if you want, and another browser will use a queue, or maybe something else entirely.  

I don't have time to write up example code right now.  Maybe later.